doc/overview/people.rst

```diff
@@ -0,0 +1,41 @@
+:orphan:
+
+.. _governance-people:
+
+Current steering council and institutional partners
+===================================================
+
+Benevolent Dictator for Life
+----------------------------
+
+Alexandre Gramfort is the Benevolent Dictator for Life (BDFL)
+
+
+Steering Council
+----------------
+
+* Alexandre Gramfort
+* Britta Westner
+* Clemens Brunner
+* Dan McCloy
+* Denis Engemann
+* Eric Larson
+* Guillaume Favelier
+* Luke Bloy
+* Mainak Jas
+* Marijn van Vliet
+* Mikołaj Magnuski
+* Richard Höchenberger
+* Robert Luke
+* Stefan Appelhoff
```

Comment:
would it be sensible to have these names as links, similar to how we do it in the Changelog? Simply by turning each list item into an RST link, and then including https://github.com/mne-tools/mne-python/blob/e0ec4a0de8d9722f2f8488797bafdf9334cb4b0a/doc/changes/names.inc#L1 somewhere